hazelcast/src/main/java/com/hazelcast/map/Immutable.java

```diff
@@ -1,4 +1,23 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package com.hazelcast.map;
 
+/**
+ * This is just a marker interface for Adobe Entities so that we can disable Copy on Read for this type
```

Comment:
Can you remove mentioning of Adobe and simply say that it is a hint that this object is effectively immutable and can be shared safely between threads?